docs/api/grpc.md

```diff
@@ -4,16 +4,110 @@ sidebar_position: 30
+2. GadgetManager:
+`GetGadgetInfo(GetGadgetInfoRequest) returns (GetGadgetInfoResponse)`: This RPC retrieves detailed information about a specific gadget, including its parameters, data sources, and other metadata.
+RunGadget(stream GadgetControlRequest) returns (stream GadgetEvent): This RPC runs a gadget (not necessarily built-in) and manages its lifecycle, similar to the RunBuiltInGadget method.
+
+## Messages
+1. `BuiltInGadgetRunRequest`:
+Describes a request to run a built-in gadget, specifying its name, category, parameters, nodes to run on, log level, and timeout.
+2. `GadgetRunRequest`:
+Similar to BuiltInGadgetRunRequest, but for non-built-in gadgets. It includes fields like imageName, paramValues, and version.
+3. `BuiltInGadgetStopRequest & GadgetStopRequest`:
+Simple messages used to stop a running gadget.
+4. `GadgetEvent`:
+Represents an event generated by a running gadget. It includes the event type, sequence number, payload, and data source ID.
+5. `BuiltInGadgetControlRequest & GadgetControlRequest`:
+These messages are used to control the execution of a gadget, allowing the client to either start or stop a gadget.
+6. `InfoRequest & InfoResponse`:
+InfoRequest is used to request information about the gadget service, and InfoResponse contains the response, including version information, a catalog, and other metadata.
+7. `DataElement & GadgetData`:
+DataElement represents a piece of data, and GadgetData aggregates these data elements, associating them with a node and a sequence number.
+8. `GadgetDataArray`:
+Represents an array of DataElement objects, useful for collecting multiple pieces of data from a gadget.
+9. `Param`:
+Describes a parameter that can be used by a gadget. It includes a key, description, default value, type hint, and other metadata.
+10. `GadgetInfo`:
+Contains detailed information about a gadget, including its name, associated image, data sources, and parameters.
+11. `Datasource`:
+Describes a data source associated with a gadget, including its ID, name, type, fields, and other metadata.
+12. `Field`:
+Represents a field within a data source, including its name, offset, size, type, and various flags.
+## Enumerations
+1. Kind
+An enumeration that defines various data types that a field can have, such as Bool, Int8, Int16, String, etc.
+## Key Concepts
+`Gadgets`: Tools or components that perform specific functions within the system. They can be built-in or custom.\
+`Data Sources`: Gadgets produce data that is structured in fields within data sources.\
+`Parameters`: Gadgets can be configured with various parameters that control their behavior.\
+`Events`: Gadgets emit events during their operation, which can be monitored and processed by clients.
 
 ## gadgettracermanager.proto
```

Comment:
I think this whole section can be dropped (or moved to an internal developer section), as this is only being used internally.